tools/abel/validators/README.md

```diff
@@ -0,0 +1,197 @@
+currently support [update
+operations](../../../ontology/docs/building_config.md#update)
+
+### Generating GUIDs
+
+There are two use cases for ABEL,
+[brownfield](../../../ontology/docs/building_config.md#update) and
+[greenfield](../../../ontology/docs/building_config.md#initialize). Currently ABEL
+only supports greenfield buildings or Building Configs under the [Initialize
+operation](../../../ontology/docs/building_config.md#initialize)
+
+#### Greenfield building
+
+A [greenfield building](../../../ontology/docs/building_config.md#initialize) 
+is one that has yet to be onboarded, and ABEL is used to 
+create a Building Config from scratch to model this building. In
+this case **GUIDs do not need to be provided except for when there are
+duplicate entity codes.** Then and only then will the user need to provide
+GUIDs for the entities with duplicate codes in every one of the ABEL
+spreadsheet tables.
+
+## Google Sheets template
+
+The [ABEL Spreadsheet Template](https://docs.google.com/spreadsheets/d/1tcLjFnHiXUT-xh5C1hRKiUVaUH_CzgSI8zFQ_B8q7vs/copy#gid=980240783) has a strict format that must be adhered to.
+
+Below is a definition of each table and column in the ABEL spreadsheet:
+- [Site](#site)
+- [Entities](#entities)
+- [Entity Fields](#entity-fields)
+- [States](#states)
+- [Connections](#connections)
+
+### Site
+
+The `Site` tab holds all information about a project site or building.
+
+`Building Code` *string* **required**
+
+A [human-readable code](../../../ontology/docs/building_config.md#identifiers) for a building. e.g. `US-SVL-TC2`
+
+`Entity Guid` *string*
+
+[UUID4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)) [identifier](../../../ontology/docs/building_config.md#identifiers) for a building.
+Unless you are onboarding a Google building, this value is generated by ABEL and does not need to be provided for initial onboarding.
+
+***Note:*** A Site should not be considered an entity and should not be listed in
+the `Entities` tab.
+
+### Entities
+
+The `Entities` tab holds all information about [reporting entities](../../../ontology/docs/building_config.md#reporting-physical-devices) and [virtual entities](../../../ontology/docs/building_config.md#virtual-devices) in a building or site. If ABEL is being used for Google then Guids are required for all entities in the `FACILTITIES` namespace which can be retrieved from DB API.
+
+`Entity Code` *string* **required**
+
+[Human-readable code](../../../ontology/docs/building_config.md#identifiers) for an entity e.g. `AHU-1001`
+
+`Entity Guid` *string*
+
+[UUID4](https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random))
+[identifier](../../../ontology/docs/building_config.md#identifiers) for an entity. **Note:** If an entity belongs to the `FACITITLIES` namespace and ABEL is
+being used or Google, then this field is **required** or else spreadsheet
```

Comment:
i didnt see a note saying abel will generate it automatically if omitted... is that mentioned somewhere?

Reply:
Yeah this is under Generating GUIDs and Greenfield building.